extensions/interactions/ItemSelectionInput/validator.js

```diff
@@ -89,15 +89,60 @@ oppia.filter('oppiaInteractiveItemSelectionInputValidator', [
         baseInteractionValidationService.getAnswerGroupWarnings(
           answerGroups, stateName));
 
-      if (!defaultOutcome ||
-          $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+      var selectedChoices = [];
+      if (maxAllowedCount === 1) {
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              var choicePreviouslySelected = (
+                selectedChoices.indexOf(ruleSpecs[j].inputs.x[k]) !== -1);
+              if (!choicePreviouslySelected) {
+                selectedChoices.push(ruleSpecs[j].inputs.x[k]);
+              }
+            }
+          }
+        }
+      }
+      if (selectedChoices.length < numChoices) {
+        if (!defaultOutcome ||
+            $filter('isOutcomeConfusing')(defaultOutcome, stateName)) {
+          warningsList.push({
+            type: WARNING_TYPES.ERROR,
+            message: (
+              'Please clarify the default outcome so it is less confusing to ' +
+              'the user.')
+          });
+        }
+      }
+      // For situations where max choices is 1,
+      // ensure that number of answers matches number of choices
+      if (maxAllowedCount === 1 && answerGroups.length > numChoices) {
```

Comment:
I thought your comments previously were that, in the case of maxAllowedCount being one, then the option of contains one, should only cover one answer choice. In that case then, does not contain one is the only option that could cover more than one answer choice. However, in that case, every answer choice chosen would result in a false. So, for example, if I had 3 answer choices and I chose a does not contain answer and selected two, I would have a choiceRules array that contains a [true, false, false]. Which would mean that I would need two more answer choices to cover the two false choices, bringing the total answers needed to 3.

If, on the other hand, I had some other combo of answers, that resulted in [true,true,true], i.e. all answer choices covered and I added another answer that used does not contain, then I would add another answer to the choiceRules array and it would now look like [true, true, true, false] which makes everything return false, but that isn't accurate, because all answer choices are, in fact covered.